### PR TITLE
Loadbalancer

### PR DESCRIPTION
**Pull request by jfryman**
This contains configurations for a central loadbalancer run on planck. Serves the following:

* https://st2.thebikeshed.io
* https://dashboard.thebikeshed.io
* https://puppycam.thebikeshed.io
* https://octoprint.thebikeshed.io

Each of these are backed by a wildcard SSL certificate, and will automatically forward HTTP -> HTTPS traffic.

This PR also contains some NAT reflection rules for OpenBSD to ensure it does the needful. When we figure out DNS, we can opt to change this. For now, it's how internal users will access the "external" resources.

Will merge once SSL cert gets issued (should be momentarily). Pushing for review/comment.

### CODE REVIEW

File: config/planck/etc/haproxy/haproxy.cfg [hunk 1]
```diff
@@ -0,0 +1,88 @@
+global
+  chroot  /var/lib/haproxy
+  daemon
+  group  haproxy
+  maxconn  4000
+  pidfile  /var/run/haproxy.pid
+  ssl-default-bind-ciphers  EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:ECDHE-RSA-AES128-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA128:DHE-RSA-AES128-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-GCM-SHA128:ECDHE-RSA-AES128-SHA384:ECDHE-RSA-AES128-SHA128:ECDHE-RSA-AES128-SHA:ECDHE-RSA-AES128-SHA:DHE-RSA-AES128-SHA128:DHE-RSA-AES128-SHA128:DHE-RSA-AES128-SHA:DHE-RSA-AES128-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA384:AES128-GCM-SHA128:AES128-SHA128:AES128-SHA128:AES128-SHA:AES128-SHA:DES-CBC3-SHA:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!MD5:!PSK:!RC4
```
Comment (jfryman) [inline config/planck/etc/haproxy/haproxy.cfg]: These are the recommend ciphers from https://cipherli.st in compat mode. Still too many odd issues with clients to go 100% restricted ciphers.

File: config/vili/etc/pf.conf [hunk 2]
```diff
@@ -51,8 +54,14 @@ pass in on em0 proto tcp from any to any port 8333 rdr-to 10.255.0.210 port 8333
 pass in on em0 proto tcp from any to any port 32400 rdr-to 10.255.2.242 port 32400
 
 # Pass HTTP and HTTPS traffic to planck load balancer
+## NAT Reflection rules exist to ensure access from inside bikeshed internal segment
 pass in on $internet_if proto tcp from any to port 80 rdr-to $loadbalancer port 80
+pass in on $bikeshed_if proto tcp from $bikeshed_if to $internet_if port 80 rdr-to $loadbalancer port 80
```
Comment (jfryman) [inline config/vili/etc/pf.conf]: These are a sort of necessary evil until we get split DNS up and going.

**Comment by jameswhite** [comment]
Is it possible to wire it up behind a proxy so that both

```https://st2.thebikeshed.io```
and
```https://thebikeshed.io/st2```

route to the same place? (for all sites)

**Comment by jfryman** [comment]
@jameswhite I mean, I could. Do you want that management overhead?

**Comment by jameswhite** [comment]
It's not urgent by any means. It's just proved useful in the past (on freyr/odin/thor/et al.) I'll wire it up later. I just didn't want any design decisions that would prevent it.

Like you can just go to https://thebikeshed.io and see all the things rather than remember or have to otherwise bookmark stuff.

**Comment by jfryman** [comment]
@jameswhite there you go, homie. https://github.com/websages/masterfiles/commit/c8462a86491ffb386dbe1b763e25badd94c75494

**Comment by jfryman** [comment]
Also wrote up some docs on how to access this. https://github.com/websages/bikeshed/pull/14

@jameswhite the `/` URLs do not seem to be working... might require a bit more playing with.

**Comment by jfryman** [comment]
Running in "production"

**Pull request merged**